src/utils/XReferralAdapter.sol

```diff
@@ -0,0 +1,92 @@
+// solhint-disable one-contract-per-file
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IVault} from "src/BaseVault.sol";
+
+interface IKernelStrategy {
+    function getHasAllocator() external view returns (bool hasAllocators);
+    function hasRole(bytes32 role, address account) external view returns (bool hasRole);
+}
+
+contract XReferralAdapter {
```

Comment:
can we call it VaultReferralAdapter ? 

this sounds cooler, but i want to keep it more standard